Ginger/GingerCore/ALM/OctaneCore.cs

```diff
@@ -226,5 +365,520 @@ public override bool IsServerConnected()
+                                    {
+                                        //change it to be string variable
+                                        stepActivity.Variables.Remove(stepActivityVar);
+                                        stepActivityVar = new VariableString();
+                                        stepActivityVar.Name = param;
+                                        ((VariableString)stepActivityVar).InitialStringValue = paramSelectedValue;
+                                        stepActivity.AddVariable(stepActivityVar);
+                                        stepActivity.AutomationStatus = eActivityAutomationStatus.Development;//reset status because flow control param was removed
+                                    }
+                                }
+                            }
+
+                            //add the variable selected value                          
+                            if (stepActivityVar is VariableSelectionList)
+                            {
+                                OptionalValue stepActivityVarOptionalVar = ((VariableSelectionList)stepActivityVar).OptionalValuesList.Where(x => x.Value == paramSelectedValue).FirstOrDefault();
+                                if (stepActivityVarOptionalVar == null)
+                                {
+                                    //no such variable value option so add it
+                                    stepActivityVarOptionalVar = new OptionalValue(paramSelectedValue);
+                                    ((VariableSelectionList)stepActivityVar).OptionalValuesList.Add(stepActivityVarOptionalVar);
+                                    if (isflowControlParam == true)
+                                        stepActivity.AutomationStatus = eActivityAutomationStatus.Development;//reset status because new param value was added
+                                }
+                                //set the selected value
+                                ((VariableSelectionList)stepActivityVar).SelectedValue = stepActivityVarOptionalVar.Value;
+                            }
+                            else
+                            {
+                                //try just to set the value
+                                try
+                                {
+                                    stepActivityVar.Value = paramSelectedValue;
+                                    if (stepActivityVar is VariableString)
+                                        ((VariableString)stepActivityVar).InitialStringValue = paramSelectedValue;
+                                }
+                                catch (Exception ex) { Reporter.ToLog(eLogLevel.ERROR, $"Method - {MethodBase.GetCurrentMethod().Name}, Error - {ex.Message}", ex); }
+                            }
+
+                            //add linked variable if needed
+                            if (string.IsNullOrEmpty(linkedVariable) == false)
```

Comment:
Codacy found an issue: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6317725)